trade-finance-manager-ui/server/controllers/deals/index.test.js

```diff
@@ -55,204 +220,195 @@ describe('controllers - deals', () => {
-          sortButtonWasClicked: false,
-          user: mockReq.session.user,
-          activeSortByField: '',
-          activeSortByOrder: 'ascending',
-        });
+      it('should redirect to GET deals with the correct query parameters', async () => {
+        await caseController.queryDeals(mockReq, mockRes);
+
+        expect(mockRes.redirect).toHaveBeenCalledWith('/deals/0?search=test');
       });
+    });
 
-      it('Not matching formId causes sortButtonWasClicked to be false', async () => {
-        const searchString = '';
+    describe('when a search is specified in the query parameters', () => {
+      const mockReq = JSON.parse(JSON.stringify(mockReqTemplate));
 
-        const mockReq = {
-          session: {
-            user: {},
-          },
-          body: {
-            formId: 'other-form',
-          },
-        };
+      mockReq.query.search = 'test';
 
-        await caseController.queryDeals(mockReq, res);
+      it('should redirect to GET deals with the correct query parameters', async () => {
+        await caseController.queryDeals(mockReq, mockRes);
 
-        expect(getDealsSpy).toHaveBeenCalled();
+        expect(mockRes.redirect).toHaveBeenCalledWith('/deals/0?search=test');
+      });
+    });
 
-        expect(res.render).toHaveBeenCalledWith('deals/deals.njk', {
-          heading: generateHeadingText(mockGetDeals.count, searchString),
-          deals: mockDeals,
-          activePrimaryNavigation: 'all deals',
-          activeSubNavigation: 'deal',
-          sortButtonWasClicked: false,
-          user: mockReq.session.user,
-          activeSortByField: '',
-          activeSortByOrder: 'ascending',
-        });
+    describe('when a search is specified in the both the request body and the query parameters', () => {
+      const mockReq = JSON.parse(JSON.stringify(mockReqTemplate));
+
+      mockReq.body.search = 'searchFromBody';
+      mockReq.query.search = 'searchFromQuery';
+
+      it('should redirect to GET deals with query parameters based on the search specified in the request body', async () => {
+        await caseController.queryDeals(mockReq, mockRes);
+
+        expect(mockRes.redirect).toHaveBeenCalledWith('/deals/0?search=searchFromBody');
       });
     });
   });
+
+  function itShouldMakeRequestsForDealsDataWithDefaultArguments(mockReq) {
+    it('should make requests to TFM API for the deals data with the default arguments', async () => {
+      await caseController.getDeals(mockReq, mockRes);
+      expect(api.getDeals).toHaveBeenCalledWith(
+        {
+          sortBy: {
+            field: 'tfm.dateReceivedTimestamp',
+            order: 'descending',
+          },
+          pagesize: 20,
+          page: 0,
+        },
+        'userToken',
+      );
+      expect(api.getAllAmendmentsInProgress).toHaveBeenCalledWith('userToken');
+    });
+  }
+
+  function itShouldRenderDealsTemplateWithDefaultArguments({ mockReq, overrideDealStage }) {
```

Comment:
What's the reason for wrapping the tests in these functions? Doesn't look like they're used in other places, wondering if they could just be `it('...`?

Reply:
The reason is simply to make the tests in this file more DRY. This (long) function is called in 3 places in this file